agent_api/src/main/java/dev/aikido/agent_api/storage/ServiceConfigStore.java

```diff
@@ -0,0 +1,76 @@
+package dev.aikido.agent_api.storage;
+
+import dev.aikido.agent_api.background.cloud.api.APIResponse;
+import dev.aikido.agent_api.background.cloud.api.ReportingApi;
+import dev.aikido.agent_api.helpers.logging.LogManager;
+import dev.aikido.agent_api.helpers.logging.Logger;
+
+import java.util.Optional;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public final class ServiceConfigStore {
+    private static final Logger logger = LogManager.getLogger(ServiceConfigStore.class);
+    private static final ServiceConfiguration config = new ServiceConfiguration();
+    private static final ReentrantReadWriteLock mutex = new ReentrantReadWriteLock();
+    private ServiceConfigStore() {
+    }
+
+    public static ServiceConfiguration getConfig() {
+        mutex.readLock().lock();
+        try {
+            return config;
+        } finally {
+            mutex.readLock().unlock();
+        }
```

Comment:
Return config can never throw? Maybe remove the try/catch?

Reply:
it's a try {} finally {}, not try {} catch{}, error still get's thrown